src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -373,16 +373,67 @@ Status OnOffServer::setOnOffValue(chip::EndpointId endpoint, chip::CommandId com
         return status;
     }
 
-    // if the value is already what we want to set it to then do nothing
-    if ((!currentValue && command == Commands::Off::Id) || (currentValue && command == Commands::On::Id))
+#ifdef MATTER_DM_PLUGIN_LEVEL_CONTROL
+    // the level control cluster supports on/off transitions or effects, e.g., controlled via
+    // the OnOffTransitionTime attribute: in a transition to "off", the level control cluster
+    // changes the CurrentLevel attribute using a timer until it reaches the minimum supported value
+    // and only then changes the OnOff value in the OnOff cluster. therefore, comparing the current
+    // OnOff state with the required state is not sufficient for OnOff cluster values that support
+    // level control:
+    // - the current state might be "On"
+    // - even though the level control cluster changes the state towards "Off"
+    // - and thus any other "On" value should prevent the cluster from turning off.
+    //
+    // this restriction does not exist for the "Off" command since level movements by definition
+    // are only available while the OnOff state is "On".
+    if (!LevelControlWithOnOffFeaturePresent(endpoint) || (command == Commands::Off::Id))
```

Comment:
It might be necessary to restrict this to lighting applications in the future. I don't, however, see how _not_ having this behavior can be useful for other applications - but I might be missing the bigger picture here.